project/src/main/java/com/google/sps/utilities/KeywordHelper.java

```diff
@@ -52,7 +57,33 @@ public void setContent(String content) {
       for (Entity entity : response.getEntitiesList()) {
         keywords.add(new Keyword(entity.getName(), entity.getSalience()));
       }
+    } catch (IOException e) {
+      System.err.println("IO Exception due to NLP library.");
     }
     return keywords;
   }
+
+  /**
+   * Finds the tokens with their basic forms in the current instance's content.
+   *
+   * @return mapping of each token in current instance's content to its basic form.
+   */
+  public Map<String, String> getTokensWithBasicForm() {
+    Map<String, String> tokenToBasicForm = new HashMap<String, String>();
+    try (LanguageServiceClient language = LanguageServiceClient.create()) {
+      Document doc = Document.newBuilder().setContent(content).setType(Type.PLAIN_TEXT).build();
+      AnalyzeSyntaxRequest request =
```

Comment:
nit: put lines 75 and 76 in their own function? I think if these details were replaced with a single named function, `getTokensWithBasicForm` would read really nicely and be closer to the same level of abstraction

Reply:
Here, in the first method we are using AnalyzeEntityRequest and in the second method we are using AnalyzeSyntaxRequest. Do you mind clarifying what you mean by same level of abstraction?

Reply:
sure! It's not really a big deal but, to me, these 2 lines make up a cohesive unit that could be abstracted away in this function so when you read this function, that part is just `getTokensList(doc)`

Reply:
where `getTokensList(doc)` makes the request and makes the response and returns response.getTokensList()

Reply:
Done - thanks for the clarification.